load_data.m

```diff
@@ -0,0 +1,65 @@
+DATA_DIR = "C:\Users\George Sykes\OneDrive - University of Leeds\Documents\year 4\project\group work\Previous data analysis\Previous Data from Leeds\bovine disc tests\*\Extract cyclic data";
+
+data = load_all_data(DATA_DIR);
+
+function data = load_all_data(data_dir)
+    files = dir(data_dir + "\*group*.csv");
+    
+    data = cell2table(cell(0,17), 'VariableNames', {'TotalTime_hrs_', 'TotalCycles', 'Step', 'Position_Linear_10_0_0_2_0__Position__mm_', 'Load_Linear_10_0_0_2_0__Load__N_', 'Displacement_Linear_10_0_0_2_0__DigitalPosition__mm_', 'SpecimenHeight_mm_', 'DerivedDiscHeight_mm_', 'DeltaH_mm_', 'DiscStress_MPa_', 'DiscStrain', 'state', 'tail' ,'disc_id', 'disc_index', 'group_id', 'run_id'});
+    
+    progress_bar = waitbar(0, "Loading data files");
+    for file_index = 1:length(files)
+        progress = file_index / length(files);
+        waitbar(progress, progress_bar, "Loading data files")
+        disc_data = load_file(files(file_index));
+        data = [data; disc_data]; 
+    end
+    close(progress_bar);
+    data.state = categorical(data.state);
+    data.tail= categorical(data.tail);
+    data.disc_id = categorical(data.disc_id);
+    data.disc_index = categorical(data.disc_index);
+    data.group_id = categorical(data.group_id);
+    data.run_id = categorical(data.run_id);
+end
+
+function disc_data = load_file(file)
+        disc_info = process_filename(file.name);
+        disc_data = readtable(strcat(file.folder + "\" + file.name));
```

Comment:
need to convert disc_data to a class that contains named parameters (or a dict/struct if matlab has them)

Reply:
I don't think we actually no need to make It unnecessarily complex, a table will do fine